frontend/src/types/media.ts

```diff
@@ -109,10 +110,15 @@ export interface ImageDimensions {
 export type AspectRatio = "square" | "intrinsic"
 
 export const isDetail = {
-  audio: (media: Media): media is AudioDetail => {
-    return media.frontendMediaType === "audio"
-  },
-  image: (media: Media): media is ImageDetail => {
-    return media.frontendMediaType === "image"
-  },
+  audio: (media: Media | null): media is AudioDetail =>
+    isMediaDetail<typeof AUDIO>(media, AUDIO),
+  image: (media: Media | null): media is ImageDetail =>
+    isMediaDetail<typeof IMAGE>(media, IMAGE),
```

Comment:
FWIW (just a nit), TypeScript can infer `T` based on the argument passed for `mediaType`, so we don't need to explicitly pass the type parameter:

```suggestion
  audio: (media: Media | null): media is AudioDetail =>
    isMediaDetail(media, AUDIO),
  image: (media: Media | null): media is ImageDetail =>
    isMediaDetail(media, IMAGE),
```

It's a shame that the return type gets flattened to `boolean` though :cry: TS doesn't transfer the `is` narrowing to the return type of `audio` :shrug: